ahnlich/ai/src/engine/ai/providers/ort.rs

```diff
@@ -0,0 +1,266 @@
+use crate::cli::server::SupportedModels;
+use crate::engine::ai::models::{InputAction, Model, ModelInput};
+use crate::engine::ai::providers::ProviderTrait;
+use crate::error::AIProxyError;
+use ahnlich_types::ai::AIStoreInputType;
+use hf_hub::{api::sync::ApiBuilder, Cache};
+use log;
+use ort::{ExecutionProviderDispatch, GraphOptimizationLevel, Session};
+
+use std::convert::TryFrom;
+use std::default::Default;
+use std::fmt;
+use std::path::{Path, PathBuf};
+use std::thread::available_parallelism;
+
+#[derive(Default)]
+pub struct ORTProvider {
+    cache_location: Option<PathBuf>,
+    cache_location_extension: PathBuf,
+    supported_models: Option<SupportedModels>,
```

Comment:
A bit confused why this would be optional

Reply:
SupportedModels shouldn't be optional, but cache_location is optional because it is usually set when `setup_provider` is called through `Model`. The location comes from the user's config.